lib/Common/Memory/Recycler.h

```diff
@@ -1966,9 +1972,12 @@ class Recycler
     {
         ObjectBeforeCollectCallback callback;
         void* callbackState;
+        void* threadContext;
+        ObjectBeforeCollectCallbackWrapper callbackWrapper;
 
         ObjectBeforeCollectCallbackData() {}
-        ObjectBeforeCollectCallbackData(ObjectBeforeCollectCallback callback, void* callbackState) : callback(callback), callbackState(callbackState) {}
+        ObjectBeforeCollectCallbackData(ObjectBeforeCollectCallbackWrapper callbackWrapper, ObjectBeforeCollectCallback callback, void* callbackState, void* threadContext) : 
+            callbackWrapper(callbackWrapper), callback(callback), callbackState(callbackState), threadContext(threadContext) {}
     };
```

Comment:
nit pick: The order of these variables are a bit inconsistent and hard to read among function SetObjectBeforeCollectCallback, struct ObjectBeforeCollectCallbackData, and constructor ObjectBeforeCollectCallbackData.

Thanks for fixing this issue thoroughly! :+1: